### ISSUE TITLE
Data format of AGORA test result

### ISSUE THREAD

**Issue by linjing7** [mks0601/Hand4Whole_RELEASE#49]
Hi, thanks for your excellent work.  

I test Hand4Whole on AGORA test set and run the `check_pred_format.py` from AGORA github to verify that the prediction results are in correct format. However, an error occurs as: `joints should be of shape (24,2) but you are providing (1, 137, 2).` I note that the save format of `joints` in `AGORA.py` is as follow: `'joints': joint_proj.reshape(1, -1, 2)`. Can I modify it as`'joints': joint_proj.reshape(-1, 2)[1:1+24, :]`  to fix the bug?

**Comment by mks0601** [comment]
That is weird. Are you testing on body-only test set?

**Comment by linjing7** [comment]
I test with the pretrained model provided in github. The result is quite curious and the F1 score is pretty low. I think there maybe something wrong with the `'joint'` I save in AGORA.py since it is used for person matching.
<img width="1115" alt="image" src="https://user-images.githubusercontent.com/70081899/201977253-3b33ba92-6296-4c56-8836-9f0e4b75bf8c.png">

**Comment by mks0601** [comment]
I think the AGORA team changed the eval codes.. Is there any clarification which 24 joints should be used?

**Comment by linjing7** [comment]
Hi, I check the evaluation code of AGORA. The shape of `gt joints` is `(127, 2) `and the first 24 joints are used for matching.

**Comment by mks0601** [comment]
Then could you try 'joints': joint_proj.reshape(-1, 2)[:24, :]?

**Comment by linjing7** [comment]
Hi, I have tried it. But unfortunately, the result keeps the same.  I check the `gt joints` in the evaluation code and find it is the same as `gt_joints_2d` from the AGORA label.

**Comment by mks0601** [comment]
In your previous answer, you used [1:1+24]. Have you tried [0:0:24]?

**Comment by linjing7** [comment]
Yes, I have tried both cases, but the result keeps the same. The evaluation code uses IoU of bbox calculated from the 24 kpts for person matching,  and I think it would not be greatly influenced by `[1:1+24]` or `[0:0:24]`.

Sorry for troubling you and I will check the bug by myself.  If I found the solution, I'll update it here.

**Comment by mks0601** [comment]
The current codes are the very ones that I used to upload results to AGORA evaluation benchmark, so I think current codes might work. I understand that AGORA dataset is kind of complicated, and required format for the evaluation is not clear :(

**Comment by linjing7** [comment]
Okay, thanks for your reply, I'll double-check my data processing of AGORA dataset and try again.

**Comment by linjing7** [comment]
@mks0601 Hi, could you please upload the AGORA_test_bbox.json again? The google link is invalid now.

**Comment by mks0601** [comment]
Could you try after deleting caches? I recently changed url, and it works for me

**Comment by linjing7** [comment]
Hi, I have deleted caches, but it does not work. Do you mean this [link](https://drive.google.com/file/d/15AV75-P4HRDDCIgDITlkdO6S3HL-g3YE/view?usp=sharing)?

**Comment by mks0601** [comment]
Ah I meant this link: https://drive.google.com/drive/folders/18CWsL28e8v50rqEbYMoU4yHHWoGJdpg_?usp=sharing
, which is from here: https://github.com/mks0601/Hand4Whole_RELEASE#data

**Comment by mks0601** [comment]
I fixed that link, too

**Comment by mks0601** [comment]
Good for you! What was the problem?

**Comment by linjing7** [comment]
Hi, @mks0601, I think there may be something wrong during my data processing of AGORA. I prepare AGORA following the instruction again and the mistake disappear.  

BTW, we don't need to run `check_pred_format.py` from AGORA github to check the data format. I directly submit the result saved by `AGORA.py` and can obtain a correct result.

**Comment by mks0601** [comment]
Got it!
